jobs/e2e-test.groovy

```diff
@@ -302,6 +305,19 @@ def runTestServer() {
       // before it does so, so it doesn't read "old" data.
       sh("rm -rf genfiles/test-reports");
 
+      tests = sh(script: "testing/runtests_server.py -n ${exec.shellEscapeList(runSmokeTestsArgs)} . ", returnStdout: true);
```

Comment:
I think you'll find it easier to use exec.outputOf() here.  Likewise below.

Reply:
I suppose that's possible. The `sh` function is pretty simple to use and comes with jenkins so has lots of great documentation and examples, whereas `exec.outputOf` is our own code with no testing or documentation that I can find. It does essentially what is done here as a convenience, but it looks to require it's arguments be provided as an array, though I never remember exactly how groovy switches between arg lists and arrays so perhaps if it's as trivial as `exec.outoutOf("testing/runtests_server.py -n ${runSmokeTestsArgs} .")` then I suppose it would be easier.

Reply:
It would be something like:
   exec.outputOf(["testing/runtests_server.py", "-n"] + runSmokeTestArgs)
It is not only easier, it is safer, because you don't have to worry about escaping.  Honestly, `sh` was a bad design decision on jenkins' part.  As you say, figuring out what `exec` does is not difficult, and it matches how exec works in most every programming language.